public/utils/contextMenu/actions.tsx

```diff
@@ -38,6 +44,8 @@ export const openContainerInFlyout = async ({
   embeddable: any;
   detectorId?: string;
 }) => {
+  const indexPatternId = embeddable.vis.data.aggs.indexPattern.id;
```

Comment:
I think you should be able to get data source id directly from the embeddable object `embeddable.vis.data.indexPattern.dataSourceRef.id`

Reply:
changed now